### ISSUE TITLE
Window is not defined error in dev logs

### ISSUE THREAD

**Issue by dml0031** [apollographql/apollo-client-nextjs#261]
After recent updates with v0.9.0 and v0.9.1 the dev console logs are being flooded with this error:

<img width="956" alt="Screenshot 2024-04-01 at 12 56 24 PM" src="https://github.com/apollographql/apollo-client-nextjs/assets/61286551/30c26cb4-7972-4ed6-a834-f839b3229294">

As far as I can tell it isn't causing any actual issues in the application, everything builds and functions as intended. My Dev console logs are just flooded with this over and over. Can provide any extra information that may be necessary.

**Comment by jerelmiller** [comment]
The 0.9 release of the next package changed exports to use [export conditions](https://nodejs.org/api/packages.html#conditional-exports). It seems from your screenshot that the [browser bundle](https://github.com/apollographql/apollo-client-nextjs/blob/a923020a39b92d62b502ec970be92a3f12fb4cd0/packages/client-react-streaming/package.json#L57) is getting pulled in for some reason on your server when it should be pulling either the [node](https://github.com/apollographql/apollo-client-nextjs/blob/a923020a39b92d62b502ec970be92a3f12fb4cd0/packages/client-react-streaming/package.json#L58) or [react-server](https://github.com/apollographql/apollo-client-nextjs/blob/a923020a39b92d62b502ec970be92a3f12fb4cd0/packages/client-react-streaming/package.json#L56) bundle. The `browser` bundle contains references to `window` which would explain your error.

Unfortunately I don't know a ton beyond that as I'm fairly unfamiliar with export conditions myself, but perhaps this is something to go on? The maintainer that knows more about this stuff should be back in the next few days so I'll see if I can get some help here when he is back.

**Comment by dml0031** [comment]
Well the providers/clients are re-used across a lot of apps in a monorepo setup. These are the only 2 imports from this package to create the re-usable clients.

Client Provider:
<img width="1274" alt="Screenshot 2024-04-01 at 1 17 02 PM" src="https://github.com/apollographql/apollo-client-nextjs/assets/61286551/3a427d53-90fd-49db-9001-4ee8ed8fd3da">

Server Provider:
<img width="817" alt="Screenshot 2024-04-01 at 1 17 36 PM" src="https://github.com/apollographql/apollo-client-nextjs/assets/61286551/bac7a76a-3e58-4dc8-aa51-4b0ff0fa5871">

And they are both exported from the same package. Figured the "use client" and "use server" directives would be enough to make sure everything gets run where its supposed to but maybe not. Not sure if this helps give any more insight.

**Comment by dml0031** [comment]
> The 0.9 release of the next package changed exports to use [export conditions](https://nodejs.org/api/packages.html#conditional-exports). It seems from your screenshot that the [browser bundle](https://github.com/apollographql/apollo-client-nextjs/blob/a923020a39b92d62b502ec970be92a3f12fb4cd0/packages/client-react-streaming/package.json#L57) is getting pulled in for some reason on your server when it should be pulling either the [node](https://github.com/apollographql/apollo-client-nextjs/blob/a923020a39b92d62b502ec970be92a3f12fb4cd0/packages/client-react-streaming/package.json#L58) or [react-server](https://github.com/apollographql/apollo-client-nextjs/blob/a923020a39b92d62b502ec970be92a3f12fb4cd0/packages/client-react-streaming/package.json#L56) bundle. The `browser` bundle contains references to `window` which would explain your error.
> 
> Unfortunately I don't know a ton beyond that as I'm fairly unfamiliar with export conditions myself, but perhaps this is something to go on? The maintainer that knows more about this stuff should be back in the next few days so I'll see if I can get some help here when he is back.

Okay so after looking into it some more, the error only appears on a page with `export const runtime = 'edge'` after removing that, the error is gone. Is there something else I should be using for the edge runtime ?

**Comment by jerelmiller** [comment]
That's helpful! I found [this discussion](https://github.com/vercel/next.js/discussions/39094) related to `edge` runtime and the `browser` export. Perhaps we need to add a `worker` exports key as mentioned by [this comment](https://github.com/vercel/next.js/discussions/39094#discussioncomment-5544248)? I'll connect with Lenz when he's back and see what the best course of action would be.

**Comment by phryneas** [comment]
I believe this will be solved by https://github.com/apollographql/apollo-client-nextjs/pull/263

Could you please give 
```
npm i @apollo/experimental-nextjs-app-support@0.0.0-commit-84f7c8d
```
a try?

**Comment by dml0031** [comment]
> I believe this will be solved by #263
> 
> Could you please give
> 
> ```
> npm i @apollo/experimental-nextjs-app-support@0.0.0-commit-84f7c8d
> ```
> 
> a try?

It does appear the issue is gone with that version!
